internal/runners/exec/exec.go

```diff
@@ -108,16 +113,20 @@ func (s *Exec) Run(params *Params, args ...string) error {
 		return locale.WrapError(err, "err_exec_env", "Could not retrieve environment information for your runtime")
 	}
 	logging.Debug("Trying to exec %s on PATH=%s", args[0], env["PATH"])
-	// Ensure that we are not calling the exec recursively
-	if _, isBeingShimmed := env[constants.ExecEnvVarName]; isBeingShimmed {
-		// Ensure that we would not call the executor recursively: The path for the executable should be different from the default bin dir
-		p := exeutils.PathForExecutable(filepath.Base(args[0]))
-		binDir := globaldefault.BinDir(s.cfg)
-		if p == binDir {
-			return locale.NewError("err_exec_recursive_loop", "Detected recursive loop while calling {{.V0}}", args[0])
-		}
+
+	recursionLvl := 0
+	lastLvl, err := strconv.ParseInt(os.Getenv(constants.ExecRecursionLevelEnvVarName), 10, 32)
+	if err != nil {
+		recursionLvl = int(lastLvl) + 1
+	}
+
+	// Report recursive execution of executor: The path for the executable should be different from the default bin dir
+	p := exeutils.FindExecutableOnOSPath(filepath.Base(args[0]))
+	binDir := filepath.Clean(globaldefault.BinDir(s.cfg))
+	if p == binDir && (recursionLvl == 1 || recursionLvl == 10 || recursionLvl == 50) {
+		return logging.Criticalf("executor recursion detected: %s (lvl=%d)", strings.Join(args, " "), recursionLvl)
```

Comment:
Why is this returning? Also as noted we just want to log this without affecting the user. So that makes it not critical.

Reply:
Additionally this should include the process args of the invoking process, as noted in my comment on the previous review. Logging this is pointless if we can't identify what led to the recursion.

Reply:
I found a package that (platform-independently) resolves the executable name of the calling process.  That might be enough to give us hints.  Getting all the arguments for a process seems to be more difficult to do (on all platforms!).

Reply:
Ok, let's start with the executable name.

Note I think we have a psutil lib in use already. Might want to see if that one can already offer this feature so we don't bring in a new package for nothing.